cql3/restrictions/single_column_restriction.hh

```diff
@@ -392,6 +393,53 @@ public:
     }
 };
 
+class single_column_restriction::LIKE final : public single_column_restriction {
+private:
+    ::shared_ptr<terminal> _value;
+    like_matcher _like_matcher;
+public:
+    LIKE(const column_definition& column_def, ::shared_ptr<terminal> value)
+            : single_column_restriction(column_def)
+          , _value(value)
+          , _like_matcher(*value->get(query_options::DEFAULT))
+    { }
```

Comment:
How can you use query_options::DEFAULT? If the LIKE pattern was a bind variable, then we won't have anything to bind to.

Reply:
btw, it presents an optimization problem: if the value was a literal, we should construct the matcher during the prepare phase (like you do here) to avoid constructing it on every query. However, LIKE is unlikely to be used with literals (we're probably searching for some user specified pattern, not an app specified pattern) and the cost of construction is likely to be amortized by many matches, since we're filtering, so it's not important to solve.

Reply:
> How can you use query_options::DEFAULT?

Because at this time I have no other `query_options` object to work with.  And when I do have it later, there is no existing API call in the base class to let me construct the `like_matcher` just once.

> If the LIKE pattern was a bind variable, then we won't have anything to bind to.

Perhaps that should be prohibited for the time being?

Reply:
It's probably the most common use case.

Reply:
I think the only option now to to create the like_matcher in is_satisfied_by(). It is sad but at least it is correct.